### PR TITLE
feat(wallet): app version check

### PR DESCRIPTION
**Pull request by doums**
# Description

Closes: 
- #3305 
- #3306

### CODE REVIEW

File: nym-wallet/src-tauri/src/operations/app/version.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        .map(|u| AppVersion {
+            current_version: u.current_version().to_string(),
+            latest_version: u.latest_version().to_owned(),
+            is_update_available: u.is_update_available(),
+        })
+        .map_err(|e| {
+            log::error!("An error ocurred while checking for app update {}", e);
+            BackendError::CheckAppVersionError
+        })?;
+    log::trace!(
```
Comment (octol) [inline nym-wallet/src-tauri/src/operations/app/version.rs]: Possibly make this debug? imho it's quite useful and not very noisy

**Review by octol — APPROVED** [review]
Rust parts looks good

**Review by gala1234 — APPROVED** [review]
I'm giving it a thumbs up, but this one has more rust code... I think it's better to wait for Fouad

**Comment by doums** [comment]
Hey, thx @gala1234  for the review. Rust code has been reviewed by Jon already, so we should be good.

**Pull request merged**